frontend/src/pages/roles/components/UserRoleRow.tsx

```diff
@@ -0,0 +1,96 @@
+import Popover from '@mui/material/Popover';
+import RoleDTO from '@shared/dtos/RoleDTO';
+import UserDTO from '@shared/dtos/UserDTO';
+import { useRef, useState } from 'react';
+import FakeProfile from '../../../utility_components/FakeProfile/FakeProfile';
+
+export function UserRoleRow({
+    user,
+    roles,
+    removeRole,
+    addRole,
+}: {
+    user: UserDTO;
+    roles: RoleDTO[];
+    removeRole: (role: RoleDTO, userId: string) => void;
+    addRole: (role: RoleDTO, userId: string) => void;
+}) {
+    const [showRoles, setShowRoles] = useState<boolean>(false);
+    const ref = useRef(null);
+
+    function availableRolesToAdd() {
+        const rolesToAdd = roles.filter(
```

Comment:
Let's break this up just for readability.

I suggest something like this:

```
    const currentRoleNames = ['Admin', 'Default', ...user.roles.map((role) => role.name)];

    const rolesToAdd = roles.filter((role) => !currentRoleNames.includes(role.name));
```

Reply:
done